app/src/main/java/org/wikipedia/edit/preview/EditPreviewFragment.java

```diff
@@ -167,19 +182,48 @@ with the original Locale (from above)
             boolean isActive = savedInstanceState.getBoolean("isActive");
             previewContainer.setVisibility(isActive ? View.VISIBLE : View.GONE);
             if (isActive) {
-                displayPreview(previewHTML);
+                displayPreviewHtml(previewHTML);
             }
         }
     }
 
+    private void displayPreviewHtml(String previewHTML) {
+        webview.loadData(previewHTML, "text/html", "UTF-8");
+        setUpWebView();
+    }
+
     public void setCustomSummary(String summary) {
         otherTag.setText(summary.length() > 0 ? summary : getString(R.string.edit_summary_tag_other));
         otherTag.setSelected(summary.length() > 0);
     }
 
     private boolean isWebViewSetup = false;
 
-    private void displayPreview(final String html) {
+    /**
+     * Fetches preview html from the modified wikitext text, and shows (fades in) the Preview fragment,
+     * which includes edit summary tags. When the fade-in completes, the state of the
+     * actionbar button(s) is updated, and the preview is shown.
+     * @param title The PageTitle associated with the text being modified.
+     * @param wikiText The text of the section to be shown in the Preview.
+     */
+    public void showPreview(final PageTitle title, final String wikiText) {
+        hideSoftKeyboard(requireActivity());
+        parentActivity.showProgressBar(true);
+        String url = model.getTitle().getWikiSite().uri() + PAGE_HTML_PREVIEW_ENDPOINT + title.getPrefixedText();
+        String postData;
+        try {
+            postData = "wikitext=" + URLEncoder.encode(wikiText, "UTF-8");
```

Comment:
We have a utility method called `UriUtil.encodeUrl()` that automatically takes care of catching unsupported encoding exceptions, which will never happen.